test/e2eTests/pages/page.js

```diff
@@ -10,16 +14,25 @@ const translator = require('./../translator');
  * This class adds the component's properties
  * to the class instance which lets tests call component methods
  * on the instance as if they were one object. See Compoent for more details.
+ *
  */
+
 class Page {
   constructor(options) {
+
+    if(!options) {
+      options={isLoaded:{ urlPart:"", element: ""}};
+    }
+
     // Make sure the components array is inialized
     options.components = options.components || [];
 
     this.options = options;
 
     // Every page should have the header component
     if (!this.options.components.includes('header')) {
+      // eslint-disable-next-line angular/log
+      console.error("no header found on page, adding...");
```

Comment:
I don't think this should be an error. If it should be an error we should probably throw an exception here since this is a development time error and should be more than just an error in the console. A bunch of failing tests will definitely catch the devs eye.

In my opinion, requiring every component to add 'header' to their list of components seems more tedious than beneficial. On the other hand, there is as least one page I know of, the lab reports page, that doesn't have the header, so maybe adding this requirement is more correct. In any case I vote to remove this change to make adding a new page a little simpler.

Reply:
The code that was already here, added the header without notifying the developer/user at all... Supposing in this in the case of the lab report, is it likely a developer who did not know about it and could have used Page, only to later realize that header was being added and causing some unexpected symptom? I think we need to make it obvious that these conveniences are happening, rather than taking it for granted till/while it bites someone.